src/services/response-service.ts

```diff
@@ -72,45 +75,106 @@ export default class ResponseService extends PercyClientService {
       return
     }
 
-    if (!this.shouldCaptureResource(rootUrl, response.url())) {
-      // Disallow remote redirects.
-      logger.debug(`Skipping [is_remote_redirect] [${width} px]: ${response.url()}`)
-      return
+    if (isRedirect) {
+      // We don't want to follow too deep of a chain
+      if (request.redirectChain().length > 4) {
+        logger.debug(`Skipping [redirect_too_deep: ${request.redirectChain().length}] [${width} px]: ${response.url()}`)
+        return
+      }
+
+      return this.handleRedirectResouce(url, response, width, logger)
     }
 
-    const localCopy = await this.makeLocalCopy(response, logger)
+    return this.handlePuppeteerResource(url, response, width, logger)
+  }
+
+  /**
+   * Handle processing and saving a resource that has a redirect chain. This
+   * will download the resource from node, and save the content as the orignal
+   * requesting url. This works since axios follows the redirect chain
+   * automatically.
+   *
+   */
+  async handleRedirectResouce(redirectedURL: string, response: puppeteer.Response, width: number, logger: any) {
+    logger.debug(`Making local copy of redirected response: ${redirectedURL}`)
+
+    const { data } = await Axios(redirectedURL, { responseType: 'text' }) as any
+    const buffer = Buffer.from(data, 'utf8')
+    const sha = crypto.createHash('sha256').update(buffer).digest('hex')
+    const localCopy = path.join(os.tmpdir(), sha)
+    const didWriteFile = this.maybeWriteFile(localCopy, buffer)
+    const { fileIsTooLarge, responseBodySize } = this.checkFileSize(localCopy)
+
+    if (!didWriteFile) {
+      logger.debug(`Skipping file copy [already_copied]: ${response.url()}`)
+    }
 
-    const responseBodySize = fs.statSync(localCopy).size
-    if (responseBodySize > Constants.MAX_FILE_SIZE_BYTES) {
-      // Skip large resources
+    if (fileIsTooLarge) {
       logger.debug(`Skipping [max_file_size_exceeded_${responseBodySize}] [${width} px]: ${response.url()}`)
       return
     }
 
-    const contentType = response.headers()['content-type']
-    const resource = this.resourceService.createResourceFromFile(url, localCopy, contentType, logger)
-    this.responsesProcessed.set(url, resource)
+    // By not setting contentType, it serves it correctly in our proxy
+    const resource = this.resourceService.createResourceFromFile(redirectedURL, localCopy, '', logger)
```

Comment:
I'm suspicious of not setting a content type 🤔 

Could we get the content type from the axios response?

Reply:
Ah yes! I need to check this, but I might have to save the file as the actual file, then use a lib to get its mime type, then sha, then send. 

UNLESS you know of a better way where I can request with axios and the content type isn’t always `text/plain` (which then fails to serve in the proxy as css). Basically is there a better way to request the content of the file, and then turn it into a buffer?

Reply:
Most libs use the file extension to determine the mime-type. So if you go that route, you might not have to write the actual filename.

As for axios, could you use `responseType: 'arraybuffer'`?

Following up with that, would `data` then already be a buffer?